lib/web/JsonpMainTemplatePlugin.js

```diff
@@ -121,15 +121,17 @@ class JsonpMainTemplatePlugin {
 						: "",
 					"script.charset = 'utf-8';",
 					`script.timeout = ${chunkLoadTimeout / 1000};`,
+					`script.src = ${mainTemplate.requireFn}.p + ${scriptSrcPath};`,
 					crossOriginLoading
-						? `script.crossOrigin = ${JSON.stringify(crossOriginLoading)};`
+						? `script.src.indexOf(window.location.origin) && script.crossOrigin = ${JSON.stringify(
+								crossOriginLoading
+						  )};`
 						: "",
 					`if (${mainTemplate.requireFn}.nc) {`,
 					Template.indent(
 						`script.setAttribute("nonce", ${mainTemplate.requireFn}.nc);`
 					),
 					"}",
-					`script.src = ${mainTemplate.requireFn}.p + ${scriptSrcPath};`,
```

Comment:
At the very least, moving this to before `script.crossOrigin` is set will make this patch *much* easier in a userland plugin.